pool/pool_machine.c

```diff
@@ -720,7 +720,23 @@ M0_INTERNAL int m0_poolmach_state_transit(struct m0_poolmach       *pm,
 					event->pe_state);
 		m0_poolmach_event_list_dump_locked(pm);
 	}
-	pm->pm_pver->pv_is_dirty = state->pst_nr_failures > 0;
+
+	/*
+	 * Skipping to mark pool version as dirty in case of dix/md pool.
```

Comment:
I believe, As per DIX redundancy policy (N+K+0) we have to mark pool version as dirty if present failures >K failures. Please correct me if i am wrong.

Reply:
>>why formulaic pool versions cannot be used for DIX pools.
I believe, modified code is formulaic pool version code. Here we are marking dirty if failures greater than K failures.

Reply:
> failures >K failures

If we have `nr_failures > K` then it means the pool version is totally dead because DIX uses 1+K-way replication. If you have 3 failures in a pool version with NKSP=1+2+0+3 then it means the whole cluster is down.
Dirty flag simply means "degraded" mode.

> why formulaic pool versions cannot be used for DIX pools.

They are not generated by Hare at the moment for "DIX" pools. They are generated only for so-called "SNS" pools.

> I believe, modified code is formulaic pool version code

The modified code has no relation to formulaic pool version code. Please take a look at https://github.com/Seagate/cortx-motr/blob/main/conf/pvers.h#L27

Just to be clear: formulaic pool versions may not be usable here due to S=0. However, the approach where we generate several pvers with different properties is what we could explore. 

We have two separate problems to solve:

1. We cannot create new indices if we don't have a clean pver. The solution here is to generate more pvers with different combinations of devices and NKSP values. What is the point of exposing pvers to S3 if we do not provide S3 with a clean pver?

2. With S=0, DIX may not be able to execute PUT and DEL operations due to the lack of spare space (it cannot preserve the 1+K replication property if some nodes are down and we have no spares). If we want to abandon the replication property in face of failures then let's modify the code pieces that assume K=S. In this case we won't be making obsolete the code paths that deal with failures and K=S. The way how we handle "S=0" would be just an additional feature of DIX.

Reply:
>If we have nr_failures > K then it means the pool version is totally dead because DIX uses 1+K-way replication. If you have 3 failures in a pool version with NKSP=1+2+0+3 then it means the whole cluster is down.

Yes, It is valid if 3-node cluster with 1+2+0.